alerts/cluster-monitoring-operator/KubeDaemonSetMisScheduled.md

```diff
@@ -0,0 +1,104 @@
+# KubeDaemonSetMisScheduled
+
+## Meaning
+
+The `KubeDaemonSetMisScheduled` alert triggers when, over a defined time interval,
+an excessive number of daemonset pods are found running on nodes where they
+should not be running. This condition typically occurs when a pod is scheduled
+on a node with a taint that doesn't have a corresponding toleration, or when a
+node is unhealthy or overloaded, which can hinder pod scheduling.
+
+## Impact
+
+Daemonset pods are running where they are not supposed to run. Excessive resource
+usage where they could be used by other apps.
+
+## Diagnosis
+
+- Check Daemonset status
+
+  ```console
+  oc -n $NAMESPACE describe daemonset $NAME
+  ```
+
+  Replace $NAME with the name of the Daemonset having the issue.
+  $NAMESPACE comes along with alert labels, replace it with respective value
+  from the alert received.
+
+- Check the status of the pods belonging to the Daemonset and look for the `Node`,
+  `Tolerations`, and `NodeSelector` fields:
+
+  ```console
+  oc -n $NAMESPACE describe pod $NAME
+  ```
+
+  Replace $NAME with the name of the pod having the issue.
+
+  - Look for the `Node`, `Node-Selectors` and `Tolerations` fields in the output.
+
+  - Check the labels for the node and verify with `NodeSelector` on the pod.
+
+    ```console
+    oc get node $NAME --show-labels
+    ```
+
+    Replace $NAME with the node name of the pod having the issue.
+
+  - Verify whether `Node-Selectors` field for the pod has one of the
+    same labels as the node
+
+    ```console
+    oc -n $NAMESPACE describe pod $NAME | grep Node-Selectors
+    ```
+
+  - Check the `Taints` for the node and verify that it matches the
+    `Tolerations` on the pod:
+
+    - Check `Taints` on the node
+
+      ```console
+      oc describe node $NAME
+      ```
+
+      Replace $NAME with the name of the node identified in the previous step.
+
+    - Check `Tolerations` on the pod
+
+      ```console
+      oc-n $NAMESPACE describe pod $NAME | grep Tolerations
+      ```
+
+      Replace $NAME with the name of the pod having the issue.
+
+    - Verify that the identified `Taints` are tolerated (identified in the
+      `Tolerations` field in the previous step).
+  
+## Mitigation
```

Comment:
I'm wondering if there's much that can be done to mitigate the alert if the affected daemonset is managed by OCP? E.g. any change to the daemonset tolerations should be reverted by the managing operator?

Reply:
Taints can be still checked if thats been tolerated or not right?